relayer/processor/path_processor.go

```diff
@@ -81,33 +84,42 @@ func (pp *PathProcessor) PathEnd2Messages(message string) SequenceCache {
 
 // Path Processors are constructed before ChainProcessors, so reference needs to be added afterwards
 // This can be done inside the ChainProcessor constructor for simplification
-func (pp *PathProcessor) SetChainProcessorIfApplicable(chainID string, chainProcessor ChainProcessor) bool {
-	if pp.pathEnd1.info.ChainID == chainID {
-		pp.pathEnd1.chainProcessor = chainProcessor
+func (pp *PathProcessor) SetChainProviderIfApplicable(chainProvider provider.ChainProvider) bool {
+	if chainProvider == nil {
+		return false
+	}
+	if pp.pathEnd1.info.ChainID == chainProvider.ChainId() {
+		pp.pathEnd1.chainProvider = chainProvider
 		return true
-	} else if pp.pathEnd2.info.ChainID == chainID {
-		pp.pathEnd2.chainProcessor = chainProcessor
+	} else if pp.pathEnd2.info.ChainID == chainProvider.ChainId() {
+		pp.pathEnd2.chainProvider = chainProvider
 		return true
 	}
 	return false
 }
 
+// ProcessBacklogIfReady gives ChainProcessors a way to trigger the path processor process
+// as soon as they are in sync for the first time, even if they do not have new messages.
+func (pp *PathProcessor) ProcessBacklogIfReady() {
+	pp.retryProcess <- struct{}{}
+}
+
 // ChainProcessors call this method when they have new IBC messages
-func (pp *PathProcessor) HandleNewMessages(chainID string, channelKey ChannelKey, messages MessageCache) {
+func (pp *PathProcessor) HandleNewMessages(chainID string, channelKey ChannelKey, cacheData ChainProcessorCacheData) {
 	if pp.pathEnd1.info.ChainID == chainID {
 		// TODO make sure passes channel filter for pathEnd1 before calling this
-		pp.handleNewMessagesForPathEnd(pp.pathEnd1, messages)
+		pp.handleNewCacheDataForPathEnd(pp.pathEnd1, cacheData)
 	} else if pp.pathEnd2.info.ChainID == chainID {
 		// TODO make sure passes channel filter for pathEnd2 before calling this
-		pp.handleNewMessagesForPathEnd(pp.pathEnd2, messages)
+		pp.handleNewCacheDataForPathEnd(pp.pathEnd2, cacheData)
 	}
 }
 
-func (pp *PathProcessor) handleNewMessagesForPathEnd(
+func (pp *PathProcessor) handleNewCacheDataForPathEnd(
```

Comment:
I would probably this method in its two calls above, since it is just one channel send.

Reply:
done